### ISSUE TITLE
linter: unsafe fixer for oxc/erasing op

### ISSUE THREAD

**Issue by camc314** [oxc-project/oxc#10833]
https://x.com/yagiznizipli/status/1919757870201032843

```ts
const output = -1 * 0
```

```
oxlint  -D erasing-op test.ts --fix
```

should not fix the above code as it is unsafe

**Comment by camc314** [comment]
this fixer should be marked as a dangerous fixer NOT a suggestion as suggestions.

suggestions are classed as "Changes that may change your program's behavior or make a change you may not want (e.g. removing console.logs). ".

**Comment by Boshen** [comment]
Should the fixer fix this case `-1 * 0` at all?

**Comment by Boshen** [comment]
> actually, maybe we shouldn't even report that case. If someone is writing `-1 * 0`, they're (probably?) doing it deliberatly to get `-0`, so we probably shouldn't report an error?

I think so, how does clippy handle this?

**Comment by camc314** [comment]
clippy checks that the `0` that we're multiplying aginst is a `Int` of value `0`:

https://github.com/rust-lang/rust-clippy//blob/bb93c23c0854bbb8aa6a523f0b4462da0f75f6da/clippy_lints/src/operators/erasing_op.rs#L42

```rs
0 * -1 // reports
0.0 * -1.0 // doesn't report
```

If it's a float, it won't report (i guess that because you can get -0).

Maybe we should just suggest replacing `-1 * 0` with just `-0` for clarity
